code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentManagerImpl.java

```diff
@@ -246,6 +246,17 @@ public void doWithLockNoResult() {
         });
     }
 
+    @Override
+    public void activateGlobalServicesForHost(long hostId) {
+        List<Service> services = sdSvc.getActiveGlobalServices();
+        for (Service service: services) {
```

Comment:
I think we should think about calling activate() in parallel here, otherwise things can get slow + if for example service1 fails to activate due to lock wait timeout, service2 activation wouldn't be attempted in this run.